pkg/util/metric/histogram_buckets.go

```diff
@@ -0,0 +1,178 @@
+// that records a quantity (nanosecond-denominated) in which most measurements
+// behave like network latencies, i.e. most measurements are in the ms to sub-second
+// range during normal operation.
+var NetworkLatencyBuckets = []float64{
+	// Generated via TestHistogramBuckets/NetworkLatencyBuckets.
+	500000.000000,    // 500µs
+	860513.842995,    // 860.513µs
+	1480968.147973,   // 1.480968ms
+	2548787.184731,   // 2.548787ms
+	4386533.310619,   // 4.386533ms
+	7549345.273094,   // 7.549345ms
+	12992632.226094,  // 12.992632ms
+	22360679.774998,  // 22.360679ms
+	38483348.970335,  // 38.483348ms
+	66230909.027573,  // 66.230909ms
+	113985228.104760, // 113.985228ms
+	196171733.362212, // 196.171733ms
+	337616984.325077, // 337.616984ms
+	581048177.284016, // 581.048177ms
+	999999999.999999, // 999.999999ms
+}
+
+// BatchProcessLatencyBuckets are prometheus histogram buckets suitable for a
+// histogram that records a quantity (nanosecond-denominated) in which most
+// measurements are in the seconds to minutes range during normal operation.
+var BatchProcessLatencyBuckets = []float64{
+	// Generated via TestHistogramBuckets/BatchProcessLatencyBuckets.
+	500000000.000000,    // 500ms
+	789604072.059876,    // 789.604072ms
+	1246949181.227077,   // 1.246949181s
+	1969192302.297256,   // 1.969192302s
+	3109764521.125753,   // 3.109764521s
+	4910965458.056452,   // 4.910965458s
+	7755436646.853539,   // 7.755436646s
+	12247448713.915894,  // 12.247448713s
+	19341270753.704967,  // 19.341270753s
+	30543892291.876068,  // 30.543892291s
+	48235163460.447227,  // 48.23516346s
+	76173362969.685760,  // 1m16.173362969s
+	120293595166.717728, // 2m0.293595166s
+	189968625172.725128, // 3m9.968625172s
+	300000000000.000183, // 5m0s
+}
+
+// LongRunningProcessLatencyBuckets are prometheus histogram buckets suitable
+// for a histogram that records a quantity (nanosecond-denominated) for
+// long-running processes (multiple minutes).
+var LongRunningProcessLatencyBuckets = []float64{
+	// Generated via TestHistogramBuckets/LongRunningProcessLatencyBuckets.
+	500000000.000000,     // 500ms
+	942961049.923126,     // 942.961049ms
+	1778351083.344248,    // 1.778351083s
+	3353831609.364442,    // 3.353831609s
+	6325065151.263324,    // 6.325065151s
+	11928580151.734879,   // 11.928580151s
+	22496372927.944168,   // 22.496372927s
+	42426406871.192848,   // 42.426406871s
+	80012898335.451462,   // 1m20.012898335s
+	150898093243.579315,  // 2m30.898093243s
+	284582048872.726685,  // 4m44.582048872s
+	536699575188.601318,  // 8m56.699575188s
+	1012173589826.278687, // 16m52.173589826s
+	1908880541934.094238, // 31m48.880541934s
+	3599999999999.998535, // 59m59.999999999s
+}
+
+// CountBuckets are prometheus histogram buckets suitable for a histogram that
+// records a quantity that is a count (unit-less) in which most measurements are
+// in the 1 to ~1000 range during normal operation.
+var CountBuckets = []float64{
```

Comment:
Here too it might make sense to encode the 1k upper limit in the name, `Count1KBuckets`, to avoid accidentally using this in an unsuitable context.